llvm/lib/ProfileData/InstrProfWriter.cpp

```diff
@@ -494,17 +494,43 @@ static uint64_t writeMemProfFrames(
 static llvm::DenseMap<memprof::FrameId, memprof::LinearFrameId>
 writeMemProfFrameArray(
     ProfOStream &OS,
-    llvm::MapVector<memprof::FrameId, memprof::Frame> &MemProfFrameData) {
+    llvm::MapVector<memprof::FrameId, memprof::Frame> &MemProfFrameData,
+    llvm::DenseMap<memprof::FrameId, memprof::FrameStat> &FrameHistogram) {
   // Mappings from FrameIds to array indexes.
   llvm::DenseMap<memprof::FrameId, memprof::LinearFrameId> MemProfFrameIndexes;
 
-  // Sort the FrameIDs for stability.
+  // Compute the order in which we serialize Frames.  The order does not matter
+  // in terms of correctness, but we still compute it for two reasons:
+  //
+  // - Stability of the output: If we sort by FrameIds at least, then the output
```

Comment:
This is largely not done afaict? I guess as a fallback tie breaker, but maybe better to mention that briefly at the end of the next paragraph?

Reply:
Fixed in the latest iteration.  It's a lot better to mention the FrameId comparison as the final tie breaker all the away at the end.